### DISCUSSION TITLE
Vague perception of ArrayableTrait::resolveFields() mechanism

### DISCUSSION THREAD

**Discussion by q4ev** [yiisoft/yii2#19729] [Q&A]
Can someone explain to me why we have `if`s in ArrayableTrait::resolveFields()?

To not throw when user requested via RESTful API non-existent fields? Why do we use keys in $this->fields() and $this->extraFields() then? Cause at the moment in current code they'll never pass these `if` checks.

In the beginning I thought it was designed to use aliases for `toArray()` function. Like return ["title"=>"name"]; So we receive result array looking like ["title" => <value of the `name` attribute>].

Please, enlighten me.

P.S. By the way if we have "*" in the `fields()` method, it'll throw Exception, thou I assume it should not.

**Comment by samdark** [comment]
You mean [this if](https://github.com/yiisoft/yii2/blob/master/framework/base/ArrayableTrait.php#L222)? That allows two input formats: `['field1', 'field2']` and `['field1' => [...], 'field2' => [...]]`.

**Reply by q4ev** [reply]
Ehm, okay. What's supposed to be in `[...]` then?

**Reply by samdark** [reply]
https://github.com/yiisoft/yii2/blob/master/framework/base/ArrayableTrait.php#L52

**Reply by q4ev** [reply]
It wouldn't work for relations. You can't `alias` them in current code. Sure, you can workaround in callback, but what if you only need rename the result key in the output?
